test/integration/mongodb-handshake/mongodb-handshake.prose.test.ts

```diff
@@ -0,0 +1,108 @@
+import { expect } from 'chai';
+
+import { getFAASEnv, MongoClient } from '../../mongodb';
+
+describe('FAAS Environment Prose Tests', function () {
+  let client: MongoClient;
+
+  afterEach(async function () {
+    await client?.close();
+  });
+
+  type EnvironmentVariables = Array<[string, string]>;
+  const tests: Array<{
+    context: string;
+    expectedProvider: string | undefined;
+    env: EnvironmentVariables;
+  }> = [
+    {
+      context: '1. Valid AWS',
+      expectedProvider: 'aws.lambda',
+      env: [
+        ['AWS_EXECUTION_ENV', 'AWS_Lambda_java8'],
+        ['AWS_REGION', 'us-east-2'],
+        ['AWS_LAMBDA_FUNCTION_MEMORY_SIZE', '1024']
+      ]
+    },
+    {
+      context: '2. Valid Azure',
+      expectedProvider: 'azure.func',
+      env: [['FUNCTIONS_WORKER_RUNTIME', 'node']]
+    },
+    {
+      context: '3. Valid GCP',
+      expectedProvider: 'gcp.func',
+      env: [
+        ['K_SERVICE', 'servicename'],
+        ['FUNCTION_MEMORY_MB', '1024'],
+        ['FUNCTION_TIMEOUT_SEC', '60'],
+        ['FUNCTION_REGION', 'us-central1']
+      ]
+    },
+    {
+      context: '4. Valid Vercel',
```

Comment:
Per my comments before, I am assuming Vercel will also have the AWS Lambda env variables present as well but would need to verify on an actual Vercel function.

Reply:
We'll follow up from the channel but regardless these are the verbatim spec prose tests

Reply:
I wanna leave the prose as is to match the spec, we can pull in an update to the spec when there is one. I did add a unit test for this case